File: problem-solving/baekjoon/2635_수_이어가기.py
# https://www.acmicpc.net/problem/2635
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    max_count = 0
    answer = []
    for second in range(1, n+1):
        numbers = [n, second]
        i = 2
        while numbers[i-2]-numbers[i-1] >= 0:
            numbers.append(numbers[i-2]-numbers[i-1])
            if max_count < len(numbers):
                max_count = len(numbers)
                answer = numbers
            i += 1
    print(max_count)
    for i in answer:
        print("{} ".format(i), end='')
    print()

File: problem-solving/baekjoon/2635_수_이어가기.go
// https://www.acmicpc.net/problem/2635
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)

	var maxCount int
	var answer []int
	for second := 1; second <= n; second++ {
		var numbers = []int{n, second}
		for i := 2; numbers[i-2]-numbers[i-1] >= 0; i++ {
			numbers = append(numbers, numbers[i-2]-numbers[i-1])
		}
		if maxCount < len(numbers) {
			maxCount = len(numbers)
			answer = numbers
		}
	}
	fmt.Fprintln(writer, maxCount)
	for i := 0; i < len(answer); i++ {
		fmt.Fprintf(writer, "%d ", answer[i])
	}
	fmt.Fprintln(writer, "")
}
